test/common/testlib.py

```diff
@@ -1829,6 +1829,8 @@ def write_file(self, path, content, append=False, owner=None, perm=None, post_re
         '''
         m = self.machine
         self.restore_file(path, post_restore_action=post_restore_action)
+        dirname = os.path.dirname(path)
+        m.execute(f"mkdir -p {dirname}")
```

Comment:
This commit could drop the `mkdir` call from test/verify/check-packagekit (`mkdir -p /etc/tracer`), packagelib.py, and check-sosreport. This would clean up the tests, and prove that this works properly.

Reply:
Done.